### PR TITLE
Implementation of translate transformation function

### PR DESCRIPTION
**Pull request by nangahamandine**
I have added the functions to translate the circle, rectangle and ellipse shapes and added a simple rectangle to be translated

### CODE REVIEW

**Comment by nangahamandine** [comment]
@nikochiko can you please verify..?

File: lib/shape.ml [hunk 1]
```diff
@@ -1,5 +1,72 @@
...
+  set_color black;
+
+  (* Create shapes *)
+  let r1 = rectangle 200 100 in
+  let r2 = translate 100 0 r1 in
+
+  (* Display shapes *)
+  show [r1; r2];
+
+  close ()
```
Comment (nikochiko) [inline lib/shape.ml:62]: Great work! 🎉

the lib is not executed by default.
as a convention is most projects, the `lib` directory is dedicated for all the *buildup* code, that is then imported into something you want to be executable.

You should instead copy this into a new file in the `examples/` directory, along with the necessary imports.

Then try building the examples directory and running that file with `dune build` and `dune exec` like mentioned in the earlier issues.

Comment (nangahamandine) [reply lib/shape.ml:62]: Okay, thank you for this! I'll try that now.

Comment (nangahamandine) [reply lib/shape.ml:62]: I have created new shape.ml and shape.mli files in the `examples` directory and copied the codes into them. 

Is that what you asked me to do or am I supposed to copy just some part of the code into a new file?

I've done the above and the result is still not displaying so I want to confirm if I'm doing it right.

**Comment by Sudha247** [comment]
Thanks for the work @nangahamandine! Could you please remove the diff from #10, and keep this PR to just the `translate` function and related changes? Then the diff will be restricted to this PR, and reviewing and merging will be easier.

**Comment by nangahamandine** [comment]
Okay, I'll do that now and make the commit.

**Comment by nangahamandine** [comment]
I've made the update. This PR now only has the implementation of the translate transformation. Is there anything else that you'd like for me to do?

**Comment by nangahamandine** [comment]
@Sudha247, @nikochiko and @AryanGodara I wish to ask if it's okay for me to proceed with working on another issue while waiting for a review on this issue?

**Comment by Sudha247** [comment]
> I wish to ask if it's okay for me to proceed with working on another issue while waiting for a review on this issue?

I'll get to reviewing this later today, I have some meetings for the next few hours. Meanwhile, please feel free to work on other things. I'd suggest perhaps building more examples with the existing `Shape` API. You can try to recreate some of the examples from #10 using `Shape`.

**Comment by nangahamandine** [comment]
> > I wish to ask if it's okay for me to proceed with working on another issue while waiting for a review on this issue?
> 
> I'll get to reviewing this later today, I have some meetings for the next few hours. Meanwhile, please feel free to work on other things. I'd suggest perhaps building more examples with the existing `Shape` API. You can try to recreate some of the examples from #10 using `Shape`.

Okay, thank you.

File: lib/shape.ml [hunk 2]
```diff
@@ -1,5 +1,72 @@
...
+
+
+
+
+
+(* open Graphics
```
Comment (Sudha247) [inline lib/shape.ml:68]: Quick note: you need to comment out the part below. You should add the `translate` function below the `ellipse` implementation.

File: lib/shape.ml [hunk 3]
```diff
@@ -1,5 +1,72 @@
...
+
+let init () =
+  open_graph " 500x500";
+  set_color black
+
+let close () =
+  ignore (read_line ());
+  close_graph ()
+
+let () =
```
Comment (Sudha247) [inline lib/shape.ml:51]: As @nikochiko mentioned, this part should be removed from here and go into `examples/`.

**Review by Sudha247 — CHANGES_REQUESTED** [review]

File: lib/shape.ml (hunk 3 above)
Comment (nangahamandine) [reply lib/shape.ml:51]: Okay, on it

File: lib/shape.ml (hunk 1 above)
Comment (nangahamandine) [reply lib/shape.ml:62]: > Great work! 🎉
> 
> Anything in `lib/` is not executed, as a convention. In most projects, the `lib` directory is dedicated for all the _buildup_ code, that is then imported into some other module that you want to be executable.
> 
> You should instead copy this into a new file in the `examples/` directory, along with the necessary imports.
> 
> Then try building the examples directory and running that file with `dune build` and `dune exec` like mentioned in the earlier issues.

I wish to ask, what file in `examples/` should I transfer this part of the code in? And what are the necessary imports that you mentioned?

Comment (nikochiko) [reply lib/shape.ml]: You can call it `translate_rectangle.ml` for this example. That is specific and makes it clear what the example is about.

**Comment by nangahamandine** [comment]
I have made the most recent commits without moving the last part to `examples/` because I'm not sure which file to add it to. 

Also, please pardon the multiple commits, I had a hard time merging the current branch with my main branch and I'm still struggling to do that.


Please let me know where in `examples/` I'm supposed to move the implementations in the last part of the code.

File: lib/shape.ml (hunk 1 above)
Comment (nangahamandine) [reply lib/shape.ml]: > You can call it `translate_rectangle.ml` for this example. That is specific and makes it clear what the example is about.

Okay, thank you

Comment (nangahamandine) [reply lib/shape.ml]: > You can call it `translate_rectangle.ml` for this example. That is specific and makes it clear what the example is about.

So for each shape's translate transformation, I add a new file for it?

**Comment by Sudha247** [comment]
Thank you for the updates @nangahamandine. But, I'm struggling a bit to review this PR as it has a lot of unrelated diff. I think you've accidentally removed the examples from `examples/` directory.

Can we please keep this PR to just addition of `translate` function and related tests. Feel free to create a new branch or PR if needed, and close this PR if it's easier for you.

**Comment by nangahamandine** [comment]
Okay, thank you. I'll definitely close this and create another branch now

**Comment by nangahamandine** [comment]
I'm closing this pull request and I've opened a new pull request in https://github.com/Sudha247/ocaml-joy/pull/20 which has a more optimized code.

**Pull request closed without merging**